tekton/integration_pipeline.go

```diff
@@ -306,5 +306,12 @@ func (r *IntegrationPipelineRun) WithIntegrationTimeouts(integrationTestScenario
 		}
 	}
 
+	// If the sum of tasks and finally timeout durations is greater than the pipeline timeout duration,
+	// increase the pipeline timeout to prevent a pipelineRun validation failure
+	if r.Spec.Timeouts.Tasks != nil && r.Spec.Timeouts.Finally != nil && r.Spec.Timeouts.Pipeline != nil &&
+		r.Spec.Timeouts.Tasks.Duration+r.Spec.Timeouts.Finally.Duration > r.Spec.Timeouts.Pipeline.Duration {
+		r.Spec.Timeouts.Pipeline = &metav1.Duration{Duration: r.Spec.Timeouts.Tasks.Duration + r.Spec.Timeouts.Finally.Duration}
```

Comment:
Can we also log some form of message like(if its not gonna be evaluated and loged by validator):
Setting pipeline timeout to sum of task timeout plus finally timeout.